lib/adapters/http/http.js

```diff
@@ -77,12 +87,12 @@ function getHost(name, opts) {
     // except for the database name) with '/'s
     uri.path = parts.join('/');
     opts = opts || {};
-    opts = utils.clone(opts);
+    opts = clone(opts);
```

Comment:
If you're wondering why I changed so many of these, it's because one of them caused a tricky bug that was hard to track down. So I decided to start cloning everywhere just as a safety precaution for the future.